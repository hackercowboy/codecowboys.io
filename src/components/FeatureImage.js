import React from 'react';
import PropTypes from 'prop-types';
import Image from './Image';

import './FeatureImage.scss';

function FeatureImage({
  src, alt, width, height,
}) {
  return (
    <div className="feature-image">
      <Image src={src} alt={alt} height={height} width={width} />
    </div>
  );
}

FeatureImage.propTypes = {
  src: PropTypes.string.isRequired,
  alt: PropTypes.string.isRequired,
  height: PropTypes.number,
  width: PropTypes.number,
};

FeatureImage.defaultProps = {
  height: undefined,
  width: undefined,
};

export default FeatureImage;
