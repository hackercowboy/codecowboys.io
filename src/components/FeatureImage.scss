.feature-image {
  display: none;
}

@media only screen and (min-width : 992px) {
  .feature-image {
    display: block;
    margin: 0 30px;
  }
}
