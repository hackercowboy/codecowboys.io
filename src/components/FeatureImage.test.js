import React from 'react';
import { shallow } from 'enzyme';

import FeatureImage from './FeatureImage';

describe('<FeatureImage/>', () => {
  it('should render', () => {
    const wrapper = shallow(<FeatureImage src="http://example.com/test.png" alt="Lorem Ipsum" />);
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
