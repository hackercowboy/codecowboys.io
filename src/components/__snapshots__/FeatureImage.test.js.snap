// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FeatureImage/> should render 1`] = `
"<div className=\\"feature-image\\">
  <Image src=\\"http://example.com/test.webp\\" alt=\\"Lorem Ipsum\\" height={[undefined]} width={[undefined]} placeholderSrc={[undefined]} visibleByDefault={false} />
</div>"
`;
